compiler/src/iree/compiler/Codegen/LLVMGPU/VectorReductionToGPU.cpp

```diff
@@ -51,6 +51,34 @@ static Value warpReduction(Location loc, OpBuilder &builder, Value input,
   return laneVal;
 }
 
+/// Emit reduction across a group for a given input.
+static Value groupReduction(Location loc, OpBuilder &builder, Value input,
+                            vector::CombiningKind kind, uint32_t size) {
+  const int warpSize = 32;
+  Value laneVal = warpReduction(loc, builder, input, kind, warpSize);
+  // if we have more than one warp, reduce across warps.
+  if (size > warpSize) {
+    uint32_t numWarp = size / warpSize;
```

Comment:
Is `size` always a multiple of 32?

Reply:
Right now yes due to restriction for the compilation pipeline. But let me add an assert.